fix: update NetworkVariable docs (#1455)
Unity-Technologies/com.unity.multiplayer.docs: docs/basics/networkvariable.md

`NetworkVariable`s are a way of synchronizing properties between servers and clients in a persistent manner, unlike [RPCs](../advanced-topics/message-system/rpc.md) and [custom messages](../advanced-topics/message-system/custom-messages.md), which are one-off, point-in-time communications that aren't shared with any clients not connected at the time of sending. `NetworkVariable`s are session-mode agnostic and can be used with either a [client-server](../terms-concepts/client-server.md) or [distributed authority](../terms-concepts/distributed-authority.md) network topology.

`NetworkVariable` is a wrapper of the stored value of type `T`, so you need to use the `NetworkVariable.Value` property to access the actual value being synchronized. A `NetworkVariable` handles value synchronization for:

* New clients joining the game (also referred to as late-joining clients)
* All connected clients during the game

> When using a [distributed authority](../terms-concepts/distributed-authority.md) network topology, A `NetworkVariable` will ensure value synchronization is handled during:

Reviewer: ```suggestion
When using a [distributed authority](../terms-concepts/distributed-authority.md) network topology, A `NetworkVariable` ensures value synchronization is handled during:
```

* Ownership changes of the associated NetworkObject

> * Object redistribution during client connection or disconnection.

Reviewer: ```suggestion
* Object redistribution during client connection or disconnection
```

- A `NetworkVariable`'s value can only be set:
    - When initializing the property (either when it's declared or within the Awake method).
    - While the associated NetworkObject is spawned (upon being spawned or any time while it's still spawned).

> - A `NetworkVariable` cannot be defined with the `static` keyword. Variables defined with the static keyword have a lifetime that lasts for the entire runtime of an application, rather than the much shorter lifetime of a `GameObject`. This results in incorrect and invalid state management where a `NetworkVariable` will not behave as expected.

Reviewer: ```suggestion
- A `NetworkVariable` can't be defined with the `static` keyword. Variables defined with the static keyword have a lifetime that lasts for the entire runtime of an application, rather than the much shorter lifetime of a `GameObject`. This results in incorrect and invalid state management where `NetworkVariable`s don't behave as expected.
```

### Initializing a NetworkVariable

> When the associated NetworkObject of a `NetworkBehaviour` with `NetworkVariable` properties is spawned, the associated `NetworkVariable` will be initialized and associated with its associated `NetworkBehaviour`. When any new client connects, it's synchronized with the current value of the `NetworkVariable` before the `NetworkBehaviour.OnNetworkSpawn` is invoked.

Reviewer: ```suggestion
When a NetworkObject with an associated `NetworkBehaviour` with `NetworkVariable` properties is spawned, the `NetworkVariable` is initialized and associated with its relevant `NetworkBehaviour`. When any new client connects, it's synchronized with the current value of the `NetworkVariable` before the `NetworkBehaviour.OnNetworkSpawn` is invoked.
```

> When the associated NetworkObject of a `NetworkBehaviour` with `NetworkVariable` properties is spawned, the associated `NetworkVariable` will also be spawned.The `NetworkVariable`'s current state (`Value`) is automatically synchronized on the client side. When any new client connects, it's synchronized with the current value of the `NetworkVariable` before the `NetworkBehaviour.OnNetworkSpawn` is invoked.

Reviewer 2: > the associated `NetworkVariable` will also be spawned. The `NetworkVariable`'s current state (`Value`) is automatically synchronized on the client side.

Perhaps:
the associated `NetworkVariable` will be initialized and associated with its associated `NetworkBehaviour`. 

Remove:

> The `NetworkVariable`'s current state (`Value`) is automatically synchronized on the client side.

This could be confusing since in a client-server network topology it will be the server spawning and doing the initial local "spawning"/synchronization of the value.

> A `NetworkVariable` will be initialized during the associated `NetworkBehaviour` initialization. The [Start](https://docs.unity3d.com/6000.0/Documentation/ScriptReference/MonoBehaviour.Start.html) and `OnNetworkSpawn` methods are invoked based on the type of NetworkObject the `NetworkBehaviour` is associated with:

Reviewer: ```suggestion
`NetworkVariable`s are initialized during the associated `NetworkBehaviour` initialization. The [Start](https://docs.unity3d.com/6000.0/Documentation/ScriptReference/MonoBehaviour.Start.html) and `OnNetworkSpawn` methods are invoked based on the type of NetworkObject the `NetworkBehaviour` is associated with:
```

You should only set the value of a `NetworkVariable` when first initializing it or if it's spawned. It isn't recommended to set a `NetworkVariable` when the associated NetworkObject isn't spawned.

### Using a NetworkVariable

> After initialization, subscribing to the `NetworkVariable.OnValueChanged` event will notify connected clients of value changes. Clients must be subscribed to this event prior to the value being changed. Typically, clients should register for `NetworkVariable.OnValueChanged` within the `OnNetworkSpawn` method.

Reviewer: ```suggestion
After initialization, subscribing to the `NetworkVariable.OnValueChanged` event notifies connected clients of value changes. Clients must be subscribed to this event prior to the value being changed. Typically, clients should register for `NetworkVariable.OnValueChanged` within the `OnNetworkSpawn` method.
```
